### PR TITLE
chore(deps): #194 make STJ conditional

### LINKED ISSUE #194: Optimise dependencies
**Is your feature request related to a problem? Please describe.**
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

**Describe the solution you'd like**
I want the package to have additional TFM'S which contains less dependencies. I.e. net core 3 & net 5, this will reduce maintenance effort. 

**Describe alternatives you've considered**
Accept the additional dependency

**Additional context**

OpenFeature.Contrib.GOFeatureFlag
- System.Text.Json

OpenFeature.Contrib.Providers.Flagsmith
- System.Text.Json

### PR DESCRIPTION
**Pull request by thompson-tomo**
## This PR

- Makes STJ only included for net framework & net standard

### Related Issues

Fixes #194

### CODE REVIEW

**Review by thomaspoignant — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

File: src/OpenFeature.Contrib.Providers.Flagsmith/OpenFeature.Contrib.Providers.Flagsmith.csproj [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard20</TargetFrameworks>
```
Comment (toddbaert) [inline src/OpenFeature.Contrib.Providers.Flagsmith/OpenFeature.Contrib.Providers.Flagsmith.csproj:4]: I'm not sure this is even a valid TFM.

**Review by toddbaert — APPROVED** [review]

File: spec
Comment (toddbaert) [inline spec:1]: Submodule update seems unrelated but isn't a problem.

**Comment by toddbaert** [comment]
@thompson-tomo Looks like there are some test-failures related to this assembly now. Do we have to change something about the test environment to support this? https://github.com/open-feature/dotnet-sdk-contrib/actions/runs/8982771607/job/24731590610?pr=196

File: src/OpenFeature.Contrib.Providers.Flagsmith/OpenFeature.Contrib.Providers.Flagsmith.csproj (hunk 1 above)
Comment (thompson-tomo) [reply src/OpenFeature.Contrib.Providers.Flagsmith/OpenFeature.Contrib.Providers.Flagsmith.csproj:4]: correct hence why i removed and instead it inherits from the master.

**Comment by thompson-tomo** [comment]
> @thompson-tomo Looks like there are some test-failures related to this assembly now. Do we have to change something about the test environment to support this? https://github.com/open-feature/dotnet-sdk-contrib/actions/runs/8982771607/job/24731590610?pr=196

My apologies i made a mistake with the conditional package management hence on net 5 was complaining about the STJ missing a feature introduced with v6 so have adjusted the condition to also include stj for net 5.

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**